yellowbrick/text/postag.py

```diff
@@ -20,125 +20,371 @@
+    the form of a list of (document) lists of (sentence) lists of 
+    (tag, token) tuples.
 
     Parameters
     ----------
+    ax : matplotlib axes
+        The axes to plot the figure on.
+    tagset: string
+        The tagset that was used to perform part-of-speech tagging.
+        Either "penn_treebank" or "universal", defaults to "penn_treebank".
+        Use "universal" if corpus has been tagged using SpaCy.
+    colors : list or tuple of colors
+        Specify the colors for each individual part-of-speech.
+    colormap : string or matplotlib cmap
+        Specify a colormap to color the parts-of-speech.
     kwargs : dict
-        Pass any additional keyword arguments to the super class.
-    cmap : dict
-        ANSII colormap
+        Pass any additional keyword arguments to the PosTagVisualizer.
 
-    These parameters can be influenced later on in the visualization
-    process, but can and should be set as early as possible.
+    Examples
+    --------
+    >>> viz = PosTagVisualizer()
+    >>> viz.fit(X)
+    >>> viz.poof()
     """
-    def __init__(self, ax=None, **kwargs):
+    def __init__(
+        self, 
+        ax=None, 
+        tagset="penn_treebank", 
+        colormap=None, 
+        colors=None, 
+        **kwargs
+    ):
+        super(PosTagVisualizer, self).__init__(ax=ax, **kwargs)
+       
+        self.tagset_names = TAGSET_NAMES
+        
+        if tagset not in self.tagset_names:
+            raise NotImplementedError(
+                "PosTagVisualizer not implemented for {} tags.".format(
+                    tagset
+                )
+            )
+        else:
+            self.tagset = tagset
+        
+        self.punct_list = PUNCT_LIST
+        self.colormap = colormap
+        self.colors = colors
+
+    def fit(self, X, y=None, **kwargs):
         """
-        Initializes the base frequency distributions with many
-        of the options required in order to make this
-        visualization work.
+        Fits the corpus to the appropriate tag map.
+        Text documents must be tokenized & tagged before passing to fit.
+
+        Parameters
+        ----------
+        X : list or generator
+            Should be provided as a list of documents or a generator
+            that yields a list of documents that contain a list of 
+            sentences that contain (token, tag) tuples.
+
+        y : ndarray or Series of length n
+            An optional array or series of target or class values for
+            instances.
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
         """
-        super(PosTagVisualizer, self).__init__(ax=ax, **kwargs)
+        # TODO: add support for other tagsets?
+        if self.tagset == "penn_treebank":
+            self.tag_map = self._penn_tag_map()
```

Comment:
`self.tag_map` is a learned parameter of this estimator (e.g. it only exists after `fit()` and a user might want access to this data). Therefore might I recommend that this property become `self.tag_map_` with the underscore suffix (e.g. following the sklearn convention) or potentially a bit more verbosely, `self.pos_tag_counts_`. This should also be identified in the `Attributes` section of the class docstring such as [this example](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/features/importances.py#L86).

Minor notes from the review process:

I had originally suggested that `self._penn_tag_map` and `self._uni_tag_map` could be candidates for `@property` decorators, but I see now that you're using these functions to avoid updating mutable data structures on successive calls to `fit` -- nice!

Reply:
Changed to `self.pos_tag_counts_`